odl/solvers/functional/functional.py

```diff
@@ -225,6 +225,48 @@ def translated(self, shift):
         """
         return FunctionalTranslation(self, shift)
 
+    def bregman(self, point, subgrad=None):
+        """Return the Bregman distance functional.
+
+        Parameters
+        ----------
+        point : element of ``functional.domain``
+            Point from which to define the Bregman distance.
+        subgrad : `Operator`, optional
+            A subgradient operator. Operator which that takes an element in
+            `functional.domain` and returns a subdifferential of the functional
+            in that point.
+            Default: ``functional.gradient``.
+
+        Returns
+        -------
+        out : `BregmanDistance`
+            The Bregman distance functional.
+
+        Notes
+        -----
+        Given a functional :math:`f`, which has a (sub)gradient
+        :math:`\partial f`, and given a point :math:`y`, the Bregman distance
+        functional :math:`D_f(\cdot, y)` in a point :math:`x` is given by
+
+        .. math::
+            D_f(x, y) = f(x) - f(y) - \\langle \partial f(y), x - y \\rangle.
```

Comment:
The notation here is non-standard. Usually, one would pick a single subgradient p from the subdifferential \partial f(y) and define D_f^p(x, y) = f(x) - f(y) - \\langle p, x - y \\rangle.